### PR TITLE
Add modern UI plugins, checks-api, update junit, remove 2.176.x & 2.190.x

### PR DESCRIPTION
**Pull request by timja**
Most of the new plugins baseline is 2.204.x, checks-api was never available on an old baseline, and all the others would need backporting, would be an absolute nightmare (this was already a pain to get all these plugins working properly).

So let's drop old baselines...

🙏  CI works across all of these

### CODE REVIEW

**Comment by timja** [comment]
weird the tests that failed are for 2.249.x

image

2.249.x uses jackson2-api-plugin 2.11.3

``
Caused by: java.io.IOException: Failed to load: JUnit Plugin (1.41)
 - Update required: Jackson 2 API Plugin (2.11.1) to be updated to 2.11.3 or higher
	at hudson.PluginWrapper.resolvePluginDependencies(PluginWrapper.java:952)
	at hudson.PluginManager$2$1$1.run(PluginManager.java:549)
	at org.jvnet.hudson.reactor.TaskGraphBuilder$TaskImpl.run(TaskGraphBuilder.java:169)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:296)
	at jenkins.model.Jenkins$5.runTask(Jenkins.java:1131)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:214)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
	... 3 more
```

```
SEVERE: Failed Loading plugin ECharts API Plugin v4.9.0-2 (echarts-api)
java.io.IOException: Failed to load: ECharts API Plugin (4.9.0-2)
 - Update required: Jackson 2 API Plugin (2.11.0) to be updated to 2.11.1 or higher
	at hudson.PluginWrapper.resolvePluginDependencies(PluginWrapper.java:952)
	at hudson.PluginManager$2$1$1.run(PluginManager.java:549)
	at org.jvnet.hudson.reactor.TaskGraphBuilder$TaskImpl.run(TaskGraphBuilder.java:169)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:296)
	at jenkins.model.Jenkins$5.runTask(Jenkins.java:1131)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:214)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Review by XiongKezhi — APPROVED** [review]
ok for the checks-api part

**Review by oleg-nenashev — APPROVED** [review]
I am fine with dropping old baselines. We are in the experimental mode, so even keeping one latest LTS is OKish. API libs is a good step towards future JUnit and Warnings Ng alignment. Approved if it works.

**Comment by timja** [comment]
> I am fine with dropping old baselines. We are in the experimental mode, so even keeping one latest LTS is OKish. API libs is a good step towards future JUnit and Warnings Ng alignment. Approved if it works.

the README says a few, we're currently keeping a lot more than a few 😂

**Comment by jglick** [comment]
Still have some problems with `jackson2-api`.

**Comment by jglick** [comment]
I am OK with removing the old lines at this point. Would want to do one last release with the current updates.

**Review by jetersen — APPROVED** [review]
LGTM.

Not sure why PCT is downgrading jackson-api in those plugins when they clearly have a higher dep in their pom.xml

**Comment by jglick** [comment]
Reproducible easily via

```bash
PLUGINS=ssh-credentials TEST=InjectedTest bash local-test.sh
```

The problem is with `io.jenkins.configuration-as-code:test-harness` as usual; it is pulling in an older version of `jackson2-api`. PCT is not prepared for this I guess.

I suspect the nonstandard reactor design in `configuration-as-code-plugin` is responsible for numerous problems. Other plugins just have a test JAR. Here we have a separate `jar`-packaging artifact with a dependency on a plugin and it seems it is a mess.

**Comment by timja** [comment]
> Reproducible easily via
> 
> ```shell
> PLUGINS=ssh-credentials TEST=InjectedTest bash local-test.sh
> ```
> 
> The problem is with `io.jenkins.configuration-as-code:test-harness` as usual; it is pulling in an older version of `jackson2-api`. PCT is not prepared for this I guess.
> 
> I suspect the nonstandard reactor design in `configuration-as-code-plugin` is responsible for numerous problems. Other plugins just have a test JAR. Here we have a separate `jar`-packaging artifact with a dependency on a plugin and it seems it is a mess.

😢 the problem is test jars can't bring in dependencies on other libraries, and the test harness is pulling in a json schema validator

**Comment by jglick** [comment]
Now there are other problems, e.g. `InjectedTest` for `git-client`

```
ECharts API Plugin version 4.9.0-2 failed to load.
 - Jackson 2 API Plugin version 2.11.0 is older than required. To fix, install version 2.11.1 or later.
```

Not exactly sure what is going on here, but it looks like PCT will need to be patched somehow.

**Comment by jglick** [comment]
Going to see if #341 removes the blockers for this.

File: bom-latest/pom.xml [hunk 1]
```diff
@@ -219,7 +254,7 @@
             <dependency>
                 <groupId>org.jenkins-ci.plugins</groupId>
                 <artifactId>junit</artifactId>
-                <version>1.29</version>
+                <version>1.41</version>
```
Comment (timja) [inline bom-latest/pom.xml:257]:
```suggestion
                <version>1.43</version>
```

**Comment by jglick** [comment]
:tada:

**Comment by jglick** [comment]
https://github.com/jenkinsci/bom/releases/tag/bom-17

**Pull request merged**